src/arena.c

```diff
@@ -1494,15 +1498,20 @@ arena_nthreads_dec(arena_t *arena, bool internal) {
 }
 
 arena_t *
-arena_new(tsdn_t *tsdn, unsigned ind, extent_hooks_t *extent_hooks) {
+arena_new(tsdn_t *tsdn, unsigned ind, const arena_config_t *config) {
 	arena_t *arena;
 	base_t *base;
 	unsigned i;
 
 	if (ind == 0) {
 		base = b0get();
 	} else {
-		base = base_new(tsdn, ind, extent_hooks);
+		extent_hooks_t *extent_hooks = config->extent_hooks ?
```

Comment:
This has one side effect: if the custom hooks passed in by the user is really NULL, then it's going to be using the default hook with this change, instead of crashing like previously.

For the existing `arenas.create`, it's a behavior change that might cover application bugs. How about we remove this check, and simply use config->extent_hooks here? I'm guessing you are trying to support the use case of setting `metadata_use_hook` w/o setting a custom hook (by using a NULL hook value in the config)? I wouldn't worry about that since it's practically the same as the all default.

Reply:
That's right - I was trying to cover the case where the config is set with NULL hooks. I've now removed this check and the corresponding test.

Thanks for the review.